homeassistant/components/tesla/device_tracker.py

```diff
@@ -27,44 +26,37 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 class TeslaDeviceEntity(TeslaDevice, TrackerEntity):
     """A class representing a Tesla device."""
 
-    def __init__(self, tesla_device, controller, config_entry):
+    def __init__(self, tesla_device, coordinator):
         """Initialize the Tesla device scanner."""
-        super().__init__(tesla_device, controller, config_entry)
-        self._latitude = None
-        self._longitude = None
+        super().__init__(tesla_device, coordinator)
         self._attributes = {"trackr_id": self.unique_id}
-        self._listener = None
-
-    async def async_update(self):
-        """Update the device info."""
-        _LOGGER.debug("Updating device position: %s", self.name)
-        await super().async_update()
-        location = self.tesla_device.get_location()
-        if location:
-            self._latitude = location["latitude"]
-            self._longitude = location["longitude"]
-            self._attributes = {
-                "trackr_id": self.unique_id,
-                "heading": location["heading"],
-                "speed": location["speed"],
-            }
 
     @property
-    def latitude(self) -> float:
+    def latitude(self) -> Optional[float]:
         """Return latitude value of the device."""
-        return self._latitude
+        location = self.tesla_device.get_location()
+        return self.tesla_device.get_location().get("latitude") if location else None
 
     @property
-    def longitude(self) -> float:
+    def longitude(self) -> Optional[float]:
         """Return longitude value of the device."""
-        return self._longitude
-
-    @property
-    def should_poll(self):
-        """Return whether polling is needed."""
-        return True
+        location = self.tesla_device.get_location()
+        return self.tesla_device.get_location().get("longitude") if location else None
 
     @property
     def source_type(self):
         """Return the source type, eg gps or router, of the device."""
         return SOURCE_TYPE_GPS
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        attr = super().device_state_attributes.copy()
+        location = self.tesla_device.get_location()
+        if location:
+            self._attributes = {
```

Comment:
We're setting the private attributes after copying the common attributes above. Is this really intended?

Reply:
That's a bug. Thanks.